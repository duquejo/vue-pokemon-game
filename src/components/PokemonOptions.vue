<template>
  <div class="options-container">
    <ul>
      <li
        v-for="option in options"
        :key="option.id"
        @click="$emit('selection', option.id)">
        {{ option.name }}
      </li>
    </ul>
  </div>
</template>

<script>
export default {
  props: {
    options: {
      type: Array,
      required: true,
    }
  },
}
</script>

<style scoped>
/* Pokemon Options */
ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  padding-top: 5px;
  width: 250px;
  text-transform: capitalize;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}

.options-container {
  display: flex;
  justify-content: center;
}
</style>