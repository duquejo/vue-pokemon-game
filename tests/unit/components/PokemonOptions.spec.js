import PokemonOptions from "@/components/PokemonOptions";
import { shallowMount } from "@vue/test-utils";
import { pokemons } from "../mocks/pokemons.mock";

describe("PokemonOptions.vue", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: {
        options: pokemons,
      },
    });
  });

  it("should match the snapshot.", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should show the 4 options as well', () => {
    const items = wrapper.findAll('li');
    expect(items).toHaveLength(4);
    expect(items[0].text()).toBe(pokemons[0].name);
    expect(items[1].text()).toBe(pokemons[1].name);
    expect(items[2].text()).toBe(pokemons[2].name);
    expect(items[3].text()).toBe(pokemons[3].name);
  });

  it('should emit \'selection\' with its args', () => {
    const [li1, li2, li3, li4] = wrapper.findAll('li');

    li1.trigger('click');
    li2.trigger('click');
    li3.trigger('click');
    li4.trigger('click');

    expect(wrapper.emitted('selection')).toHaveLength(4);

    expect(wrapper.emitted('selection')[0]).toEqual([pokemons[0].id]);
    expect(wrapper.emitted('selection')[1]).toEqual([pokemons[1].id]);
    expect(wrapper.emitted('selection')[2]).toEqual([pokemons[2].id]);
    expect(wrapper.emitted('selection')[3]).toEqual([pokemons[3].id]);
  });
});
