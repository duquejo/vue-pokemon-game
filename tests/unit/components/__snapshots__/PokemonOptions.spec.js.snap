// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonOptions.vue should match the snapshot. 1`] = `
<div class="options-container">
  <ul>
    <li>pikachu</li>
    <li>charmander</li>
    <li>ivysaur</li>
    <li>voltorb</li>
  </ul>
</div>
`;
